### PR TITLE
Reacts to the update in the orchestrator-interface and updates certif…

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Reacts to the update in the orchestrator-interface and updates certifier cert.

- Adds test cases.
- Fixes tox issues

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -911,6 +934,56 @@ def test_given_magma_service_enabled_when_install_then_nothing_done(
...
+                "certifier_pem_certificate": "certifier_pem_certificate_content",
+                "orchestrator_address": "orchestrator.com",
+                "orchestrator_port": "42",
+                "bootstrapper_address": "bootstrapper.com",
+                "bootstrapper_port": "42",
+                "fluentd_address": "fluentd.com",
+                "fluentd_port": "42",
+            },
+        )
+        patch_remove.assert_not_called()
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:960]:
```suggestion
        self.assertNotIn(call().unlink(), patch_path.mock_calls)
```

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -911,6 +934,56 @@ def test_given_magma_service_enabled_when_install_then_nothing_done(
...
+
+    @patch("charm.Path")
+    @patch("subprocess.run")
+    @patch("os.remove")
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:964]:
```suggestion
```

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -911,6 +934,56 @@ def test_given_magma_service_enabled_when_install_then_nothing_done(
...
+                "certifier_pem_certificate": "certifier_pem_certificate_content",
+                "orchestrator_address": "orchestrator.com",
+                "orchestrator_port": "42",
+                "bootstrapper_address": "bootstrapper.com",
+                "bootstrapper_port": "42",
+                "fluentd_address": "fluentd.com",
+                "fluentd_port": "42",
+            },
+        )
+        patch_remove.assert_called()
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:985]:
```suggestion
        self.assertIn(call().unlink(), patch_path.mock_calls)
```

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: tests/unit/test_charm.py (hunk 3 above)
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: tests/unit/test_charm.py (hunk 2 above)
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: tests/unit/test_charm.py (hunk 1 above)
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: tests/unit/test_charm.py [hunk 4]
```diff
@@ -911,6 +934,56 @@ def test_given_magma_service_enabled_when_install_then_nothing_done(
             ]
         )
 
+    @patch("charm.Path")
+    @patch("subprocess.run")
+    @patch("os.remove")
+    def test_given_certifier_pem_not_stored_when_certifier_pem_changed_then_remove_agw_certs_not_called(  # noqa: E501
+        self, patch_remove, patch_subprocess_run, patch_path
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: tests/unit/test_charm.py [hunk 5]
```diff
@@ -911,6 +934,56 @@ def test_given_magma_service_enabled_when_install_then_nothing_done(
             ]
         )
 
+    @patch("charm.Path")
+    @patch("subprocess.run")
+    @patch("os.remove")
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: tests/unit/test_charm.py [hunk 6]
```diff
@@ -11,7 +11,30 @@
...
+
+    def mkdir(self):
+        pass
+
+    def write_text(self, _):
+        pass
+
+    def read_text(self):
+        pass
+
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply tests/unit/test_charm.py]: ✅

File: src/charm.py [hunk 7]
```diff
@@ -289,6 +293,26 @@ def _is_valid_interface(self, interface_name: str, new_interface_name: str) -> b
...
+
+    @staticmethod
+    def _remove_agw_cert_files():
+        for file in [
+            "/var/opt/magma/gateway.crt",
+            "/var/opt/magma/gateway.key",
+            "/var/opt/magma/gw_challenge.key",
+        ]:
+            try:
+                os.remove(file)
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply src/charm.py]: ✅

File: src/charm.py [hunk 8]
```diff
@@ -7,6 +7,7 @@
 import ipaddress
 import json
 import logging
+import os
```
[the remark this answers is not in the document]
Comment (saltiyazan) [reply src/charm.py]: ✅

**Review by Gmerold — APPROVED** [review]
You can make it a regular PR now ;)

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**